codec/dagcbor/unmarshal.go

```diff
@@ -14,14 +14,29 @@ import (
 )
 
 var (
-	ErrInvalidMultibase = errors.New("invalid multibase on IPLD link")
+	ErrInvalidMultibase         = errors.New("invalid multibase on IPLD link")
+	ErrAllocationBudgetExceeded = errors.New("message structure demanded too many resources to process")
+)
+
+const (
+	mapEntryGasScore  = 8
```

Comment:
how are you deriving these? is it a rough approximation of the overhead?

Reply:
Yep.

Fairly arbitrarily derived.  I figure it takes -- very, very roughly -- at least this many bytes in memory to add another map entry to memory.

Is this number wrong?  Absolutely.  I have no idea how many bytes it takes to add a map entry in memory.  It's a property of the golang native map implementation, and varies based on the size of the map, which is not something I'm particularly interested in predicting or making specific claims about... and then, even that is only in some Nodes.  It's actually zero in some nodes (e.g. codegen'd structs).  Or who knows what in some other Node implementation: it's an interface users can supply their own of, after all.

It's sufficient to have _a_ limit; I'm not sure it's necessarily important for the limit to be easy to intuit.